File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
import java.lang.Math;

class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 0;
        int maxRound = 0;
        
        while(n>1){
            n /= 2;
            maxRound += 1;
        }
        
        int div;
        for(int i=0; i<maxRound; i++){
            div = (int)Math.pow(2,i+1);
            if(((a-1)/div) == ((b-1)/div)){
                answer = i+1;
                break;
            }
        }        

        return answer;
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
import math
def solution(n,a,b):
    answer = 0
    
    idx = 1
    a, b = a-1, b-1
    while True:
        div = int(math.pow(2,idx))
        
        if((a//div) == (b//div)):
            answer = idx
            break
        else :
            idx += 1
    

    return answer